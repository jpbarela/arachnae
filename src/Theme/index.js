// @flow
export type ButtonThemeType = {
  primary: string,
};

export type ThemeType = {
  background: string,
  button: ButtonThemeType,
  color: string,
  fontFamily: string,
};

export const defaultTheme: ThemeType = {
  button: {
    primary: "blue",
  },
  color: "black",
  background: "white",
  fontFamily: "Helvetica, sans-serif",
};
