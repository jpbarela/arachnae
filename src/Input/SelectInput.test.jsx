// @flow
import * as React from "react";
import { render } from "@testing-library/react";
import { SelectInput } from "./index";
import { ThemeProvider } from "react-jss";
import { defaultTheme } from "../Theme";

describe("SelectInput", () => {
  it("renders", () => {
    const container = render(
      <ThemeProvider theme={defaultTheme}>
        <SelectInput
          options={[
            { value: "1", name: "One" },
            { value: "2", name: "Two" },
          ]}
        />
      </ThemeProvider>
    );
    expect(container.baseElement.firstChild).toMatchSnapshot();
  });
});
