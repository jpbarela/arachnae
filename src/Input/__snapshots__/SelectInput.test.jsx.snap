// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput renders 1`] = `
<div>
  <div
    class="inputContainer-0-2-1"
  >
    <select
      class="select-0-2-2"
    >
      <option
        disabled=""
        value=""
      />
      <option
        value="1"
      >
        One
      </option>
      <option
        value="2"
      >
        Two
      </option>
    </select>
  </div>
</div>
`;
